### ISSUE TITLE
Hide configuration lifecycle

### ISSUE THREAD

**Issue by jerben** [oxidizing/sihl#410]
## Goal

`core_configuration.ml` can be used to read .env files and store the values into the env variables. Afterwards the env variables can be read. At the moment `core_app.ml` takes care of reading .env files, so you always have to keep that in mind when calling `Sihl.Configuration.read_string`. `core_configuration` should do that itself.

## Todo

Make the file reading lazy and make sure it is done only once .

- add internal state to check whether .env file was read
- whenever a `read_*` is called and the value is not memoized, read the .env file into env variables if not already done
- when reading .env file into env variables, set internal state to keep track of file ready
- expose function `read_env_file` that does all that and run it in the `run` function in `core_app`, this adds some predictability since we force the configuration to be ready now
